counsel.el

```diff
@@ -1432,11 +1427,12 @@ INITIAL-INPUT can be given as the initial minibuffer input."
   (setq counsel-git-grep-cmd
         (ivy-read "cmd: " counsel-git-grep-cmd-history
                   :history 'counsel-git-grep-cmd-history))
-  (setq counsel-git-grep-cmd-history
-        (delete-dups counsel-git-grep-cmd-history))
-  (unless (ivy-state-dynamic-collection ivy-last)
-    (setq ivy--all-candidates
-          (all-completions "" 'counsel-git-grep-function))))
+  (let ((repo-size (funcall counsel--git-grep-count-func)))
```

Comment:
Shouldn't the call to `counsel--git-grep-count-func` come after the call to `unless` further down?

```el
(setq counsel-git-grep-cmd-history
      (delete-dups counsel-git-grep-cmd-history))
(unless (ivy-state-dynamic-collection ivy-last)
  (let* ((repo-size (funcall counsel--git-grep-count-func))
         (coll (lambda (str _pred _action)
                 (counsel-git-grep-function str repo-size))))
    (setq ivy--all-candidates (all-completions "" coll))))
```